src/Common/CgroupsMemoryUsageObserver.h

```diff
@@ -9,50 +9,59 @@
 namespace DB
 {
 
-/// Periodically reads the current memory usage from Linux cgroups.
-/// You can specify soft or hard memory limits:
-/// - When the soft memory limit is hit, drop jemalloc cache.
-/// - When the hard memory limit is hit, update MemoryTracking metric to throw memory exceptions faster.
+/// Does two things:
+/// 1. Periodically reads the memory usage of the process from Linux cgroups.
+///    You can specify soft or hard memory limits:
+///    - When the soft memory limit is hit, drop jemalloc cache.
+///    - When the hard memory limit is hit, update MemoryTracking metric to throw memory exceptions faster.
+/// 2. Periodically reads the the maximum memory available to the process (which can change due to cgroups settings).
```

Comment:
I think we should mention that the actual goal of reloading the server configuration is to re-configure the internal memory tracker with updated settings `max_server_memory_usage` and `max_memory_usage_for_user`